core/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

```diff
@@ -59,10 +58,8 @@ object RedundantBraces extends Rewrite {
           case (Lit(value: String), arg @ Term.Name(name)) =>
             val openBrace = prevToken(arg.tokens.head)
             val closeBrace = nextToken(arg.tokens.head)
-            (openBrace, closeBrace, value.headOption) match {
-              case (LeftBrace(), RightBrace(), Some(c)) if !isAlphaNumericOrUnderscore(c) =>
-                builder += Patch(openBrace, closeBrace, name)
-              case (LeftBrace(), RightBrace(), None) =>
+            (openBrace, closeBrace) match {
+              case (LeftBrace(), RightBrace()) if !isIdentifierAtStart(value) =>
```

Comment:
can we move the `!isIdentifierAtStart(value)` guard to the `case Lit(value...` line?